test/extended/registry/signature.go

```diff
@@ -0,0 +1,120 @@
+		g.By("building an signer image that know how to sign images")
+		_, err := oc.Run("new-build").Args("--docker-image", "openshift/origin:latest", "--to", "signer:latest", "-D", "-").InputString(`FROM openshift/origin:latest
+RUN yum install -y skopeo && yum clean all && mkdir -p gnupg && chmod -R 0777 /var/lib/origin && \
+echo $'%echo Generating a basic OpenPGP key ...\n\
+Key-Type: DSA \n\
+Key-Length: 1024 \n\
+Subkey-Type: ELG-E \n\
+Subkey-Length: 1024 \n\
+Name-Real: Joe Tester \n\
+Name-Comment: with stupid passphrase \n\
+Name-Email: joe@foo.bar \n\
+Expire-Date: 0 \n\
+Creation-Date: 2017-01-01 \n\
+%commit \n\
+%echo done \n'\
+>> dummy_key.conf`).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = exutil.WaitForAnImageStreamTag(oc, oc.Namespace(), "signer", "latest")
+		containerLog, _ := oc.Run("logs").Args("builds/signer-1").Output()
+		e2e.Logf("signer build logs:\n%s\n", containerLog)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("looking up the openshift registry URL")
+		registryURL, err := imagesutil.GetDockerRegistryURL(oc)
+		signerImage := fmt.Sprintf("%s/%s/signer:latest", registryURL, oc.Namespace())
+		signedImage := fmt.Sprintf("%s/%s/signed:latest", registryURL, oc.Namespace())
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("obtaining bearer token for the test user")
+		user := oc.Username()
+		token, err := oc.Run("whoami").Args("-t").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("granting the image-signer role to test user")
+		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:image-signer", oc.Username()).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// TODO: The test user needs to be able to unlink the /dev/random which is owned by a
+		// root. This cannot be done during image build time because the /dev is plugged into
+		// container after it starts. This SCC could be avoided in the future when /dev/random
+		// issue is fixed in Docker.
+		g.By("granting the anyuid scc to test user")
+		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "anyuid", oc.Username()).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("preparing the image stream where the signed image will be pushed")
+		_, err = oc.Run("create").Args("imagestream", "signed").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("granting the image-auditor role to test user")
+		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:image-auditor", oc.Username()).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		pod, err := exutil.NewPodExecutor(oc, "sign-and-push", signerImage)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Generate GPG key
+		// Note that we need to replace the /dev/random with /dev/urandom to get more entropy
+		// into container so we can successfully generate the GPG keypair.
+		g.By("creating dummy GPG key")
+		out, err := pod.Exec("rm -f /dev/random; ln -sf /dev/urandom /dev/random && " +
```

Comment:
I'm not saying you can't do this, I'm just expressing my level of sadness with this hack, sigh...